vyper/semantics/validation/utils.py

```diff
@@ -456,9 +459,22 @@ def validate_expected_type(node, expected_type):
             types_str = sorted(str(i) for i in given_types)
             given_str = f"{', '.join(types_str[:1])} or {types_str[-1]}"
 
+        suggestions_str = ""
+        if (
+            len(expected_type) == 1
+            and isinstance(expected_type[0], AddressDefinition)
+            and len(given_types) == 1
+            and isinstance(given_types[0], Bytes20Definition)
+        ):
+            suggestions_str = f"Did you mean {checksum_encode(node.value)}?"
+
         # CMC 2022-02-14 maybe TypeMismatch would make more sense here
         raise InvalidType(
-            f"Expected {expected_str} but literal can only be cast as {given_str}", node
+            (
```

Comment:
the exception string here is a tuple instead of a regular string, ex.
```
vyper.exceptions.InvalidType: ('Expected address but literal can only be cast as bytes20. Did you mean 0x3cD751E6b0078Be393132286c442345e5DC49699?',)
  contract "repros/3062.vy", line 1:25 
  ---> 1 FOO: constant(address) = 0x3cd751e6b0078be393132286c442345e5dc49699
  --------------------------------^
       2
```